### PR TITLE
Change the limitLabel defaultProps and remove the 0 from numberOfItemsLeft property

### PR DESCRIPTION
**Pull request by alycecristines**
## Description 📄

When using this component without passing anything in the limitLabel property the text undefined was
showing and this PR fixes this problem. Also removed the padding 0 from the numberOfItemsLeft property. 

## Screenshots 📸
| Before |  After | 
| ------ | ------ | 
|    image    |        image    |

### CODE REVIEW

**Review by ericcleao — APPROVED** [review]

File: packages/yoga/src/Result/native/Details.jsx [hunk 1]
```diff
@@ -83,7 +83,7 @@ const ResultDetails = ({
                   <>
                     <Separator />
                     <StyledText variant={variant}>
-                      + {numberOfItemsLeft.toString().padStart(2, '0')}
+                      + {numberOfItemsLeft.toString()}
                       {limitLabel !== '' && ` ${limitLabel}`}
```
Comment (ericcleao) [inline packages/yoga/src/Result/native/Details.jsx:87]: Why do not change to:
```jsx
{!!limitLabel ? '' : `{limitLabel}`}
```
Or just

```jsx
{!!limitLabel && `{limitLabel}`}
```
Once we should not show nether `null`, empty or `undefined` as limit label.

Comment (ericcleao) [reply packages/yoga/src/Result/native/Details.jsx:87]: Thinking in cases that the `limitLabel` prop is set as `null` or `undefined` for instance (not just the default prop value).

File: packages/yoga/src/Result/native/Details.jsx [hunk 2]
```diff
@@ -83,7 +83,7 @@ const ResultDetails = ({
                   <>
                     <Separator />
                     <StyledText variant={variant}>
-                      + {numberOfItemsLeft.toString().padStart(2, '0')}
+                      + {numberOfItemsLeft.toString()}
```
Comment (leonardokl) [inline packages/yoga/src/Result/native/Details.jsx:86]:
```suggestion
                      + {numberOfItemsLeft}
```

**Review by leonardokl — APPROVED** [review]

**Comment by coelhucas** [comment]
I think this component could get snapshots for other "variants" of its props, for instance: this change should break a snapshot in order to let us know that it had a visual regression (removing the leading zeros) so we could re-generate the snapshot with confidence that it would catch visual issues.

**Review by katharinedepaulo — APPROVED** [review]

**Pull request merged**